File: VisitBosnia/VisitBosnia.Model/AppUserRole.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VisitBosnia.Model
{
    public class AppUserRole
    {
        public int Id { get; set; }
        public int AppUserId { get; set; }
        public int RoleId { get; set; }

        public virtual AppUser AppUser { get; set; } = null!;
        public virtual Role Role { get; set; } = null!;
    }
}

File: VisitBosnia/visit_bosnia_mobile/lib/model/roles/appuser_role.dart
import '../appUser/app_user.dart';
import 'role.dart';

class AppUserRole {
  int? id;
  int? appUserId;
  int? roleId;
  AppUser? appUser;
  Role? role;

  AppUserRole({this.id, this.appUserId, this.appUser, this.role, this.roleId});

  AppUserRole.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    appUserId = json['appUserId'];
    roleId = json['roleId'];
    appUser =
        json['appUser'] != null ? new AppUser.fromJson(json['appUser']) : null;
    role = json['role'] != null ? new Role.fromJson(json['role']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['appUserId'] = this.appUserId;
    data['roleId'] = this.roleId;
    if (this.appUser != null) {
      data['appUser'] = this.appUser!.toJson();
    }
    if (this.role != null) {
      data['role'] = this.role!.toJson();
    }
    return data;
  }
}
